ionoscloud/resource_server_boot_device_selection.go

```diff
@@ -99,23 +110,29 @@ func resourceServerBootDeviceSelectionRead(ctx context.Context, d *schema.Resour
 }
 
 func resourceServerBootDeviceSelectionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	ss := cloudapiserver.Service{Client: meta.(services.SdkBundle).CloudApiClient, Meta: meta, D: d}
 	dcId := d.Get("datacenter_id").(string)
 	serverId := d.Get("server_id").(string)
 
+	ss := cloudapiserver.NewUnboundService(serverId, meta)
+
 	if d.HasChange("boot_device_id") {
-		_, n := d.GetChange("boot_device_id")
-		bootDeviceId := n.(string)
-		if err := ss.UpdateBootDevice(ctx, dcId, serverId, bootDeviceId); err != nil {
+		bootDeviceIDValue, bootDeviceIDOk := d.GetOk("boot_device_id")
+		if !bootDeviceIDOk {
+			if err := ss.PxeBoot(ctx, dcId, serverId); err != nil {
+				return diag.FromErr(fmt.Errorf("error while performing pxe boot: %s, serverId: %s, dcId: %s", err.Error(), serverId, dcId))
+			}
+		}
+		bootDeviceID := bootDeviceIDValue.(string)
```

Comment:
if you needed the else above, don't you need the else here?

Reply:
fixed ([555645e](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/521/commits/555645efa0ea6f7cdbb227cfb1aae3fb448e341d))